components/datetime/src/options/mod.rs

```diff
@@ -0,0 +1,14 @@
+pub mod components;
+pub mod style;
+
+#[derive(Debug)]
+pub enum DateTimeFormatOptions {
+    Style(style::Bag),
+    Components(components::Bag),
```

Comment:
I didn't know you were doing skeletons in this PR.  Skeletons are complicated and you don't have enough logic to resolve them correctly.  I would rather stick with dateStyle/timeStyle and have more time to discuss skeletons.

Reply:
What about we keep the API, without functionality?
Showing what this is going to look like.
We have other things that don't yet work (no real data provider yet).

Reply:
I think the API needs a lot of work.

Reply:
@zbraniecki to file a follow-up.

Reply:
But this is not about skeletons, is it?
Is it just about bag of options, ECMA 402 style.

Reply:
I'm using the word "skeleton" interchangeably with "ECMA-402-style bag of options that could be transformed to a skeleton string for the purposes of date pattern resolution".

This is out of scope of this thread, but, since you seem interested: Yes, we should eventually support this API, but I think, if possible, skeleton resolution should live in the [wrapper layer](https://github.com/unicode-org/icu4x/blob/master/docs/wrapper-layer.md).  If we can avoid shipping the datetime pattern generator in the general case, that will be a huge win for code size.

Please wait for Zibi to open the follow-up issue and we can continue discussion in there.

Reply:
https://github.com/unicode-org/icu4x/issues/273